src/index.ts

```diff
@@ -86,6 +89,94 @@ const {grpc} = new gax.GrpcClient();
  * @property {Constructor} [promise] Custom promise module to use instead of
  *     native Promises.
  */
+export interface Client {
+  //tslint: disable-next-line: no-any
+  [k: string]: any;
+  servicePath?: string;
+  sslCreds?: ChannelCredentials;
+  port?: number;
+}
+export interface ClientConfig {
+  [k: string]: string|{}|Function|Client|undefined;
```

Comment:
I would start this one over, and extend `GoogleAuthOptions` as it will have most oof these fields